test/unittest/unittest_nntrainer_models.cpp

```diff
@@ -580,7 +589,9 @@ class nntrainerModelTest
   int getIteration() { return iteration; };
   nntrainer::TensorDim getLabelDim() { return label_dim; }
 
-  bool shouldCompare() { return options & ModelTestOption::COMPARE; }
+  bool shouldCompare() {
+    return (options & ModelTestOption::COMPARE) == ModelTestOption::COMPARE;
```

Comment:
I wonder about this change,` ModelTestOption::COMPARE` is 1 << 0 so it should work though.

Reply:
Later, we might fine-grain COMPARE to COMPARE_FOR and COMPARE_BACK. This will be needed for resnet kind of models where the error in derivative is significantly outside the error threshold. Hence added this.

Reply:
Alright :)